scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -52,20 +55,41 @@ abstract class ScanResultsStorage {
         var storage: ScanResultsStorage = NoStorage()
 
         /**
-         * Given a [config] for a known storage backend, configure it as the current one.
+         * Configure the [ScanResultsStorage]. If [config] does not contain a storage configuration by default a
+         * [FileBasedStorage] using a [XZCompressedLocalFileStorage] as backend is configured.
          */
-        fun configure(config: Any) {
-            when (config) {
-                is FileBasedStorageConfiguration -> configure(config)
-                is PostgresStorageConfiguration -> configure(config)
-                else -> throw IllegalArgumentException("Unsupported configuration type '$config'.")
+        fun configure(config: ScannerConfiguration) {
+            val configuredStorages = listOfNotNull(
+                config.fileBasedStorage,
+                config.postgresStorage
+            )
+
+            require(configuredStorages.size <= 1) {
+                "Only one scan results storage may be configured."
             }
+
+            val storageConfig = configuredStorages.firstOrNull()
+
+            when (storageConfig) {
+                is FileBasedStorageConfiguration -> configure(storageConfig)
+                is PostgresStorageConfiguration -> configure(storageConfig)
+                null -> configureDefaultStorage()
```

Comment:
Nit: Maybe make this case the first, as it's the default?

Reply:
Done.